src/data/feed/coinbase.js

```diff
@@ -0,0 +1,75 @@
+(function(fc) {
+    'use strict';
+
+    // https://docs.exchange.coinbase.com/#market-data
+
+    fc.data.feed.coinbase = function() {
+
+        var product = 'BTC-USD',
+            start = null,
+            end = null,
+            granularity = null;
+
+        var coinbase = function(cb) {
+            var params = [];
+            if (start != null) {
+                params.push('start=' + start.toISOString());
+            }
+            if (end != null) {
+                params.push('end=' + end.toISOString());
+            }
+            if (granularity != null) {
+                params.push('granularity=' + granularity);
+            }
+            var url = 'https://api.exchange.coinbase.com/products/' + product + '/candles?' + params.join('&');
+            d3.json(url, function(error, data) {
+                if (error) {
+                    return cb(error);
```

Comment:
Oh yes, it's all over the place. I can't say I like it though, returning something from a function for the convenience of a slightly more concise implementation seems wrong to me.

Although I did try to get a `+` through in my PR ;-)